### PR TITLE
Use single quotes around passwords to avoid shell interpretation

### PR DESCRIPTION
**Pull request by stdietrich**
Special characters in the password might be interpreted by the shell. This might lead to a failed registration at RHEL.
By using single quotes around the password, this will be avoided.

### CODE REVIEW

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
makes sense visually... any RH people care to test?

File: provisioning_templates/snippet/redhat_register.erb [hunk 1]
```diff
@@ -142,18 +142,18 @@ name: redhat_register
...
     <% end %>
   <% end %>
   <% if host_param('subscription_manager_username') && host_param('subscription_manager_password') %>
     <% if host_param('subscription_manager_pool') %>
-      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password="<%= host_param('subscription_manager_password') %>"
+      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password='<%= host_param("subscription_manager_password") %>'
       subscription-manager attach --pool="<%= host_param('subscription_manager_pool') %>"
     <% else %>
-      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password="<%= host_param('subscription_manager_password') %>" --auto-attach
+      subscription-manager register --name="<%= @host.name %>" --username="<%= host_param('subscription_manager_username') %>" --password='<%= host_param("subscription_manager_password") %>' --auto-attach
```
Comment (ares) [inline provisioning_templates/snippet/redhat_register.erb]: Looks good to me but if you're touching this, would you mind also fixing name and username? I guess there are more strict validations for these, but still...

**Comment by stdietrich** [comment]
I have added single quotes around the user- and hostnames and updated the branch accordingly.

**Review by ekohl — APPROVED** [review]
Looks good. @ares can you have another look?

**Comment by ares** [comment]
Tested what I could and works well, thanks @stdietrich, there's another call on line 167 but it does not contain username/password so that should be OK. Merging.

**Pull request merged**